.github/workflows/pyccel_lint.yml

```diff
@@ -39,9 +42,8 @@ jobs:
         shell: bash
       - name: Collect diff information
         run: |
-          BASE_BRANCH=$GITHUB_BASE_REF
           git fetch
-          git diff origin/${BASE_BRANCH}..HEAD --no-indent-heuristic --unified=0 --output=pull_diff.txt --no-color
+          git diff origin/${{ inputs.base }}..HEAD --no-indent-heuristic --unified=0 --output=pull_diff.txt --no-color
```

Comment:
could you explain this change.

Reply:
Previously `$BASE_BRANCH` was defined 2 lines higher as `BASE_BRANCH=$GITHUB_BASE_REF`. The variable `$GITHUB_BASE_REF` is only defined when the test is triggered by a pull request event (open/synchronize/mark_as_draft/etc) as documented [here](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables)) which means that it is not defined when the test is triggered via a commit using `/bot run lint` or `/bot mark as ready`. To fix this I use an input which is defined from a bot output. The bot uses the github CLI to investigate the PR and work out what the base branch is.

I removed the intermediate variable `BASE_BRANCH` as I can't see what it adds